clash-cores/src/Clash/Cores/Xilinx/Ethernet/Gmii/BlackBoxes.hs

```diff
@@ -0,0 +1,129 @@
+      gmiiTxEr :<
+      lvdsOutP :<
+      lvdsOutN :<
+      gmiiRxD :<
+      gmiiRxDv :<
+      gmiiRxEr :<
+      clk125 :<
+      rst125 :<
+      statusVector :<
+      compName :< _
+      = ((0 :: Int)...)
+    in InlineYamlPrimitive [Verilog, SystemVerilog] [__i|
+      BlackBox:
+        name: Clash.Cores.Xilinx.Ethernet.Gmii.gmiiSgmiiBridgeBb
+        kind: Declaration
+        template: |-
+          wire ~GENSYM[lvdsOutP][#{lvdsOutP}];
+          wire ~GENSYM[lvdsOutN][#{lvdsOutN}];
+          wire [7:0] ~GENSYM[gmiiRxD][#{gmiiRxD}];
+          wire ~GENSYM[gmiiRxDv][#{gmiiRxDv}];
+          wire ~GENSYM[gmiiRxEr][#{gmiiRxEr}];
+          wire ~GENSYM[pmaClk125][#{clk125}];
+          wire ~GENSYM[pmaRst125][#{rst125}];
+          wire [15:0] ~GENSYM[pmaStatusVector][#{statusVector}];
+
+          ~INCLUDENAME[0] ~GENSYM[gig_ethernet_pcs_pma][#{compName}] (
+            .refclk625_p(~ARG[#{smiiClkP}]),
+            .refclk625_n(~ARG[#{smiiClkN}]),
+            .reset(~ARG[#{rst}]),
+            .speed_is_100(0),
+            .speed_is_10_100(0),
+            .signal_detect(~ARG[#{signalDetect}]),
+            .configuration_vector(~ARG[#{pmaConfig}]),
+            .an_adv_config_vector(~ARG[#{pmaAnAdvancedConfig}]),
+            .an_restart_config(~ARG[#{restartAn}]),
+            .rxp(~ARG[#{lvdsInP}]),
+            .rxn(~ARG[#{lvdsInN}]),
+            .gmii_txd(~ARG[#{gmiiTxD}]),
+            .gmii_tx_en(~ARG[#{gmiiTxDv}]),
+            .gmii_tx_er(~ARG[#{gmiiTxEr}]),
+            .txp(~SYM[#{lvdsOutP}]),
+            .txn(~SYM[#{lvdsOutN}]),
+            .gmii_rxd(~SYM[#{gmiiRxD}]),
+            .gmii_rx_dv(~SYM[#{gmiiRxDv}]),
+            .gmii_rx_er(~SYM[#{gmiiRxEr}]),
+            .clk125_out(~SYM[#{clk125}]),
+            .rst_125_out(~SYM[#{rst125}]),
+            .status_vector(~SYM[#{statusVector}])
+          );
+          assign ~RESULT =
+            { ~SYM[#{lvdsOutP}]
+            , ~SYM[#{lvdsOutN}]
+            , ~SYM[#{gmiiRxD}]
+            , ~SYM[#{gmiiRxDv}]
+            , ~SYM[#{gmiiRxEr}]
+            , ~SYM[#{clk125}]
+            , ~SYM[#{rst125}]
+            , ~SYM[#{statusVector}]
+            };
+        includes:
+          - extension: clash.tcl
+            name: gig_ethernet_pcs_pma
+            format: Haskell
+            templateFunction: #{tfName}
+
+          |]
+
+binaryTclTF :: TemplateFunction
+binaryTclTF = TemplateFunction used valid tclTemplate
+ where
+  used = [0..11] -- use all arguments
+  valid = const True -- There are no configuration options to check
+
+tclTemplate :: Backend s => BlackBoxContext -> State s Doc
+tclTemplate bbCtx
+  | [compName] <- bbQsysIncName bbCtx
+  =
+  let
+    bbText = [__di|
+      namespace eval $tclIface {
+        variable api 1
+        variable scriptPurpose createIp
+        variable ipName {#{compName}}
+        proc createIp {ipName0 args} {
+          create_ip -name gig_ethernet_pcs_pma -vendor xilinx.com -library ip -version 16.2 \\
```

Comment:
Hmmmm. Actually, I've decided I would prefer 80 column Tcl... but it doesn't matter because it boils down to the same thing in this case! X-D

```haskell
          create_ip -name gig_ethernet_pcs_pma -vendor xilinx.com -library ip \\
            -version 16.2 -module_name $ipName0
```